framework/aster-frame/src/sync/mutex.rs

```diff
@@ -41,7 +41,7 @@ impl<T> Mutex<T> {
     }
 
     /// Release the mutex and wake up one thread which is blocked on this mutex.
-    fn unlock(&self) {
+    pub fn unlock(&self) {
```

Comment:
I see, thank you for the clarification. It's important to keep the unlock method private to prevent potential misuse.

Instead of directly calling unlock on the Mutex,  I handled the unlocking of the mutex by dropping the MutexGuard itself in the newer version, as dropping the MutexGuard will release the lock.